src/qml/RelationCombobox.qml

```diff
@@ -184,9 +184,10 @@ Item {
           id: delegateRect
 
           property int idx: index
-          property string itemText: featureListModel.searchTerm != '' ? displayString.replace(new RegExp('(' + featureListModel.searchTerm + ')', "i"), '<span style="text-decoration:underline;' + Theme.toInlineStyles({
+          readonly property string encodedDisplayString = displayString.replaceAll('&', '&amp;').replaceAll('<', '&lt;').replaceAll('>', '&gt;').replaceAll('"', '&quot;').replaceAll("'", '&#039;')
+          property string itemText: featureListModel.searchTerm != '' ? encodedDisplayString.replace(new RegExp('(' + featureListModel.searchTerm + ')', "i"), '<span style="text-decoration:underline;' + Theme.toInlineStyles({
```

Comment:
This regexp should work _if_ the replace allows for long-ahead and look-behind:

```suggestion
          property string itemText: featureListModel.searchTerm != '' ? encodedDisplayString.replace(new RegExp('(?!=&[a-z]*)(' + featureListModel.searchTerm + ')(?![a-z]*;)', "i"), '<span style="text-decoration:underline;' + Theme.toInlineStyles({
```

Proof of logic:

![image](https://github.com/user-attachments/assets/a9b031cb-e399-4e70-a1d0-6fbb57a4939f)